File: ImperatorToCK3Tests/MapperTests/DeathReasonMapper/DeathReasonMappingTests.cpp
#include "gtest/gtest.h"
#include "Mappers/DeathReasonMapper/DeathReasonMapping.h"
#include <sstream>



TEST(Mappers_DeathReasonMappingTests, ck3ReasonDefaultsToNullopt) {
	std::stringstream input;

	const mappers::DeathReasonMapping theMapping(input);

	ASSERT_FALSE(theMapping.ck3Reason);
}


TEST(Mappers_DeathReasonMappingTests, ck3ReasonCanBeSet) {
	std::stringstream input;
	input << "= { ck3 = ck3Trait }";

	const mappers::DeathReasonMapping theMapping(input);

	ASSERT_EQ("ck3Trait", theMapping.ck3Reason);
}


TEST(Mappers_DeathReasonMappingTests, imperatorReasonsDefaultToEmpty) {
	std::stringstream input;

	const mappers::DeathReasonMapping theMapping(input);

	ASSERT_TRUE(theMapping.impReasons.empty());
}


TEST(Mappers_DeathReasonMappingTests, imperatorReasonsCanBeSet) {
	std::stringstream input;
	input << "= { imp = reason_dumb imp = reason_bear }";

	const mappers::DeathReasonMapping theMapping(input);

	ASSERT_EQ(2, theMapping.impReasons.size());
	ASSERT_TRUE(theMapping.impReasons.contains("reason_dumb"));
	ASSERT_TRUE(theMapping.impReasons.contains("reason_bear"));
}

File: ImperatorToCK3.UnitTests/Mappers/DeathReason/DeathReasonMappingTests.cs
using commonItems;
using ImperatorToCK3.Mappers.DeathReason;
using Xunit;

namespace ImperatorToCK3.UnitTests.Mappers.DeathReason;

public class DeathReasonMappingTests {
	[Fact]
	public void CK3ReasonDefaultsToNull() {
		var reader = new BufferedReader("");
		var mapping = new DeathReasonMapping(reader);
		Assert.Null(mapping.Ck3Reason);
	}
	[Fact]
	public void CK3ReasonCanBeSet() {
		var reader = new BufferedReader("= { ck3 = ck3Trait }");
		var mapping = new DeathReasonMapping(reader);
		Assert.Equal("ck3Trait", mapping.Ck3Reason);
	}
	[Fact]
	public void ImperatorReasonsDefaultToEmpty() {
		var reader = new BufferedReader("");
		var mapping = new DeathReasonMapping(reader);
		Assert.Empty(mapping.ImperatorReasons);
	}
	[Fact]
	public void ImperatorReasonsCanBeSet() {
		var reader = new BufferedReader("= { ir=reason_dumb ir=reason_bear }");
		var mapping = new DeathReasonMapping(reader);
		Assert.Collection(mapping.ImperatorReasons,
			item => Assert.Equal("reason_bear", item),
			item => Assert.Equal("reason_dumb", item)
		);
	}
}